discord_signup.py

```diff
@@ -0,0 +1,204 @@
+# This is a test file for a user sign-up flow.
+# It requires the bot to be alive at all times, so I have not merged it into the main script (yet?)
+
+import configparser
+import os
+import re
+from typing import Callable, Optional
+import asyncio
+import discord
+from discord.ext import commands
+
+ONE_MINUTE = 60
+SIGNUP_TIMEOUT = ONE_MINUTE * 10
+
+# Read the settings.cfg file
+config_path = os.path.join(os.path.dirname(__file__), "settings.cfg")
+config = configparser.ConfigParser()
+config.read(config_path)
+
+# Add sections to config if they are missing
+if "DISCORD_ID_BY_EMAIL" not in config:
```

Comment:
should we extract loading the config into a method?

Reply:
👍